frontend/src/app/testQueue/TestCardForm/TestCardForm.tsx

```diff
@@ -323,7 +328,8 @@ const TestCardForm = ({
         deviceTypeId: state.deviceId,
         specimenTypeId: state.specimenId,
         dateTested: state.dateTested,
-        results: doesDeviceSupportMultiplex(state.deviceId, devicesMap)
+        // Should we filter results to only send results supported by the selected device?
```

Comment:
could you elaborate a bit more about how this described state differs from what we're already doing? In the case where `doesDeviceSupportNonCovid` and we're not already doing filtering, presumably `state.testResults` only has defined values for the diseases the devices support. Are we returning nulls / undefineds for the other diseases that we'd want to potentially filter out?

Reply:
I think my question in that comment might actually be addressed by clearing the state of test results whenever a device changes. On the old test card, if you marked Flu A as positive and then changed the device to a COVID only device, it would technically still have the Flu A value in the test results state which would just get filtered out while submitting